src/protocol/PublishRequest.js

```diff
@@ -44,8 +45,9 @@ class PublishRequest extends WebsocketRequest {
             ...super.toObject(),
```

Comment:
- Let's change all messages to be versioned arrays of the following form:

`[version, type, ...typeSpecificFields]`

- Introduce new `protocolVersion` `1`
- `WebsocketRequest` can be renamed to `ControlMessage` 
- Need to maintain backwards compatibility in both serializing and serializing old message versions
- Extend the list of message type codes that already exists (runs from 0-7 for different response messages) to include the different request types.